### PR TITLE
Maintainance: Refactor building iPad's play control toolbar

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/1023.

Fixes a review comment about improving the code which generates iPad's play control toolbar.

## Summary for release notes

Maintenance: Refactor building iPad's play control toolbar

### CODE REVIEW

**Comment by wutschel** [comment]
@kambala-decapitator, this is what you had in mind?

File: XBMC Remote/NowPlaying.m [hunk 1]
```diff
@@ -2422,33 +2423,33 @@ - (void)buildIpadPlaylistToolbar {
...
+    // Align buttons around this center
+    CGFloat buttonCenterY = CGRectGetHeight(playlistToolbarView.frame) / 2;
     
-    // Center button is playpause
-    CGFloat centerX = playlistToolbarView.center.x - CGRectGetMinX(playlistToolbarView.frame);
-    UIButton *button = [playlistToolbarView viewWithTag:TAG_ID_PLAYPAUSE];
-    button.center = CGPointMake(centerX, button.center.y);
+    // Define the list of playcontrol buttons (top to down = left to right)
+    NSArray *toolbarButtonList = @[
+        @{@"buttonTag": @(TAG_SHUFFLE),         @"originY": @(buttonCenterY + SHUFFLE_REPEAT_VERTICAL_PADDING)},
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:2431]: we can simplify the data structure a bit:
```objc
NSDictionary<NSNumber*, NSNumber*> dic = @{
  @(TAG_SHUFFLE): @(buttonCenterY + SHUFFLE_REPEAT_VERTICAL_PADDING),
  // ...
};
[dic enumerate...];
```

**Comment by kambala-decapitator** [comment]
> @kambala-decapitator, this is what you had in mind?

basically yes, but with dictionary it'll be a bit simpler :)

File: XBMC Remote/NowPlaying.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m:2431]: I tried following, but the order of the buttons is not correct as the dictionary does not keep the assignment order:

```
NSDictionary<NSNumber*, NSNumber*> *toolbarButtonList = @{
    @(TAG_SHUFFLE): @(buttonCenterY + SHUFFLE_REPEAT_VERTICAL_PADDING),
    @(TAG_ID_STOP): @(buttonCenterY),
    @(TAG_SEEK_BACKWARD): @(buttonCenterY),
    @(TAG_ID_PREVIOUS): @(buttonCenterY),
    @(TAG_ID_PLAYPAUSE): @(buttonCenterY),
    @(TAG_ID_NEXT): @(buttonCenterY),
    @(TAG_SEEK_FORWARD): @(buttonCenterY),
    @(TAG_ID_TOGGLE): @(buttonCenterY),
    @(TAG_REPEAT): @(buttonCenterY + SHUFFLE_REPEAT_VERTICAL_PADDING),
};
...
[toolbarButtonList enumerateKeysAndObjectsUsingBlock:^(NSNumber *key, NSNumber *obj, BOOL *stop) {
    UIButton *button = [playlistToolbarView viewWithTag:[key intValue]];
    button.center = CGPointMake(startX, [obj floatValue]);
    startX += buttonPadding;
}];
```
Anything I am doing wrong here?

Comment (kambala-decapitator) [reply XBMC Remote/NowPlaying.m:2431]: Sorry, didn't notice that the loop body depends on the order of buttons (in the original code it wasn't so). Yes, then array is needed.

**Pull request merged**